src/lib/address/address.ts

```diff
@@ -1,496 +1,70 @@
-          if (error.message === DANGEROUS_ETH_URL_ERROR) throw error
-          // address, not url, don't log
-          if (error.message.includes('Not an Ethereum URI')) break
-        }
-        console.error(error)
-      }
-      break
-    case btcChainId:
-    case bchChainId:
-    case dogeChainId:
-    case ltcChainId:
-      try {
-        const urnScheme = CHAIN_ID_TO_URN_SCHEME[inputChainId]
-        const parsedUrl = bip21.decode(urlOrAddress, urnScheme)
-        return {
-          assetId: inputAssetId,
-          maybeAddress: parsedUrl.address,
-          chainId: inputChainId,
-          ...(parsedUrl.options?.amount
-            ? { amountCryptoPrecision: bnOrZero(parsedUrl.options.amount).toFixed() }
-            : {}),
-        }
-      } catch (error) {
-        if (error instanceof Error) {
-          // address, not url, don't log
-          if (error.message.includes('Invalid BIP21 URI')) break
-        }
-        console.error(error)
-      }
-      break
-    default:
-      return { assetId: inputAssetId, chainId: inputChainId, maybeAddress: urlOrAddress }
-  }
-
-  return { assetId: inputAssetId, chainId: inputChainId, maybeAddress: urlOrAddress }
-}
-
-export const parseMaybeUrl = async ({
-  urlOrAddress,
+export const parseAddress = async ({
+  address,
 }: {
-  urlOrAddress: string
-}): Promise<ParseMaybeUrlResult> => {
+  address: string
+}): Promise<ParseAddressResult> => {
   // Iterate over supportedChainIds
   for (const chainId of knownChainIds) {
     try {
-      const maybeUrl = parseMaybeUrlWithChainId({ chainId, urlOrAddress })
-      const isValidUrl = maybeUrl.maybeAddress !== urlOrAddress
-
-      const adapter = getChainAdapterManager().get(chainId)
-
-      if (!adapter) throw new Error('Adapter not found')
-
-      const defaultAssetId = adapter.getFeeAssetId()
-      // Validation succeeded, and we now have a ChainId
-      if (isValidUrl) {
-        const finalAssetId = maybeUrl.assetId || defaultAssetId
-        return {
-          chainId,
-          value: urlOrAddress,
-          assetId: finalAssetId,
-          amountCryptoPrecision: maybeUrl.amountCryptoPrecision,
-        }
-      }
-
-      // Validation was unsuccessful, but this may still be a valid address for this adapter
-      const isValidAddress = await validateAddress({ chainId, maybeAddress: urlOrAddress })
+      const isValidAddress = await validateAddress({ chainId, maybeAddress: address })
       if (isValidAddress) {
+        const adapter = getChainAdapterManager().get(chainId)
+        if (!adapter) continue
+
+        const defaultAssetId = adapter.getFeeAssetId()
         return {
           chainId,
-          value: urlOrAddress,
+          value: address,
           assetId: defaultAssetId,
         }
       }
     } catch (error) {
-      // We want this actual error to be rethrown as it's eventually user-facing
-      if (error instanceof Error && error.message === DANGEROUS_ETH_URL_ERROR) throw error
```

Comment:
This still exists, but now in `parseUrlDirect()` - no URL concerns here anymore.